.github/workflows/test.yml

```diff
@@ -11,24 +11,45 @@ concurrency:
   cancel-in-progress: true
 
 jobs:
+  build-openassetio:
+    name: Build OpenAssetIO
+    runs-on: ubuntu-latest
+    container:
+      image: ghcr.io/openassetio/openassetio-build
+    steps:
+      - uses: actions/checkout@v3
+      - name: Build
+        uses: ./.github/build_openassetio
+
   test:
     name: Test-Resolver
     runs-on: ubuntu-latest
+    needs: build-openassetio
     container:
       image: aswf/ci-vfxall:2022-clang13.1
     steps:
       - uses: actions/checkout@v3
 
-      - name: Build and install (Docker)
-        run: >
-          cmake -S . -B build &&
-          cmake --build build &&
+      - name: Get OpenAssetIO
+        uses: actions/download-artifact@v3
+        with:
+          name: OpenAssetIO Build
+          path: ./openassetio-build
+
+      - name: Build and install Resolver
+        run: |
+          cmake -S . -DCMAKE_PREFIX_PATH="./openassetio-build" -B build
+          cmake --build build
           cmake --install build
 
       - run: |
           python -m pip install -r tests/requirements.txt
 
       - name: Test
-        run: >
+        run: |
+          export PXR_PLUGINPATH_NAME=$GITHUB_WORKSPACE/build/dist/resources/plugInfo.json
+          export LD_LIBRARY_PATH=/usr/local/lib:$GITHUB_WORKSPACE/openassetio-build/lib64
+          export OPENASSETIO_DEFAULT_CONFIG=$GITHUB_WORKSPACE/tests/resources/openassetio.conf
+          export PYTHONPATH=/usr/local/lib/python/:$GITHUB_WORKSPACE/openassetio-build/lib/python3.9/site-packages
```

Comment:
It's because the vfxall platform has a somewhat odd installation of python/USD.

In the `/usr/local/lib` directory, you'll find a `python3.9` dir, which has all the regular python libs in it, which is on the pythonpath. However, adjacent to that is this `python` directory, which contains only the `pxr` library, which isn't on the pythonpath by default.

Reply:
Right, got you, can we add that as a comment then so we know why (and is it supposed to be in front of ours?)

Reply:
Same for `LD_LIBRARY_PATH`?

Reply:
`LD_LIBRARY_PATH` is different but similar. David found this post
https://unix.stackexchange.com/questions/115129/why-does-root-not-have-usr-local-in-path

Potentially we could do `sudo -u aswfuser python -m pytest`, but i'm not sure I see that as altogheter much of a different approach, and the fact that it's finally working now makes me hesitant to jump back into path wrangling.

Will for sure add comments explaining these two things.

Reply:
> and is it supposed to be in front of ours?

I didn't think the ordering would matter, but if it's traditional to put the more application focused ones in the front of the path i'll just swap em around.

Reply:
Things in front are loaded in preference to ones on the back, when the same thing is found, so it depends on what the required behaviour there is, usually its that the stuff you're adding is more important.